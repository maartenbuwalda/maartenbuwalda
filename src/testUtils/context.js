const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
