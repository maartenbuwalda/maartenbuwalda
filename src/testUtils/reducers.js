export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
