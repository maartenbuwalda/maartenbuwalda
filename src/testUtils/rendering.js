import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
