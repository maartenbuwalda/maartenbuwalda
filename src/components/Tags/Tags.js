import React from 'react'
import styled, { css } from 'styled-components'
import { ColorContext } from '../../context/color'
import PropTypes from 'prop-types'
import colors from '../../constants/colors'

const Tags = ({ list, inverted }) => (
  <ColorContext.Consumer>
    {({ theme }) => {
      return <StyledTags inverted={inverted} theme={theme}>
        {list.map(tag => <span key={tag}>{tag}</span>)}
      </StyledTags>
    }}
  </ColorContext.Consumer>
)

Tags.propTypes = {
  list: PropTypes.array,
  inverted: PropTypes.bool,
}

const StyledTags = styled.div`
  display: flex;
  margin: .5rem 0;
  
  span {
    transition: background-color .3s;
    font-size: .8rem;
    text-transform: uppercase;
    margin-right: .5rem;
    ${({ theme, inverted }) => {
    if (inverted) {
      return css`
        color: ${theme.hex.colorRegular};
      `
    } else {
      return css`
        padding: .3rem .5rem;
        border-radius: .5rem;
        color: ${colors.fadedWhite};
        background-color: ${theme.hex.colorRegular};
      `
    }
  }}
  }
`

export default Tags
