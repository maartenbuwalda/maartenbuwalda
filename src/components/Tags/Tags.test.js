import React from 'react'
import Tags from './Tags'
import { componentMounts } from '../../testUtils/rendering'

describe('Tags component', () => {
  const component = <Tags list={['test']} />
  componentMounts(component)
})
