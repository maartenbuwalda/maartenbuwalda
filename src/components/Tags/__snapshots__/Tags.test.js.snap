// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags component mounts correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: .5rem 0;
}

.c0 span {
  -webkit-transition: background-color .3s;
  transition: background-color .3s;
  font-size: .8rem;
  text-transform: uppercase;
  margin-right: .5rem;
  margin-bottom: .5rem;
  padding: .3rem .5rem;
  border-radius: .5rem;
  color: rgba(255,255,255,0.7);
}

<Tags
  list={
    Array [
      "test",
    ]
  }
>
  <styled.div
    theme={
      Object {
        "hex": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
        "rgb": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: .5rem 0;
  
  span {
    transition: background-color .3s;
    font-size: .8rem;
    text-transform: uppercase;
    margin-right: .5rem;
    margin-bottom: .5rem;
    ",
              [Function],
              "
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "hex": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
          "rgb": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
        }
      }
    >
      <div
        className="c0"
      >
        <span
          key="test"
        >
          test
        </span>
      </div>
    </StyledComponent>
  </styled.div>
</Tags>
`;
